### PR TITLE
Single stateroot sender

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

#1922 originally, but some additional fixes were also needed.

### CODE REVIEW

File: pkg/services/stateroot/validators.go [hunk 1]
```diff
@@ -92,10 +96,31 @@ func (s *service) signAndSend(r *state.MPTRoot) error {
...
+	}
+	s.relayExtensible(ir.myVote)
+	delay := firstVoteResendDelay
+	if ir.retries > 0 {
+		delay = s.timePerBlock << ir.retries
+	}
+	_ = time.AfterFunc(delay, func() {
+		s.sendVote(ir)
+	})
+	ir.retries++
```
Comment (fyrchik) [inline pkg/services/stateroot/validators.go:127]: The mutex is taken on read only.

File: pkg/services/stateroot/service.go [hunk 2]
```diff
@@ -118,6 +118,14 @@ func (s *service) OnPayload(ep *payload.Extensible) error {
...
 			return nil
 		}
+		s.srMtx.Lock()
+		ir, ok := s.incompleteRoots[sr.Index]
+		if ok {
+			ir.Lock()
+			ir.isSent = true
+			ir.Unlock()
+		}
+		s.srMtx.Unlock()
```
Comment (fyrchik) [inline pkg/services/stateroot/service.go:128]: Can we move this unlock before the `if`?

File: pkg/services/stateroot/validators.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/services/stateroot/validators.go:128]: Fixed.

File: pkg/services/stateroot/service.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/services/stateroot/service.go]: Fixed.

**Review by fyrchik — APPROVED** [review]

**Pull request merged**